x-pack/legacy/plugins/canvas/public/lib/es_service.js

```diff
@@ -40,4 +41,9 @@ export const getIndices = () => {
       });
     })
     .catch(err => notify.error(err, { title: `Couldn't fetch Elasticsearch indices` }));
-};
+
+export const getDefaultIndex = () =>
+  savedObjectsClient
```

Comment:
It's probably an inexpensive call, but would it be worth caching the result so we wouldn't be making this call multiple times?

Reply:
Yeah, I talked to Clint previously about storing it in transient state and only updating it as needed.

I'm wondering if we could just make this call once and load it into Canvas as part of the initial state and not worry about updating it once it changes. I'm not sure how other parts of Kibana handle default index changes, but I think it'd fair to expect the user to refresh Canvas in order for the changes to apply.

Reply:
I think I'm going to leave it as is for now. The call is inexpensive, and Elasticsearch handles caching pretty well. If it presents to be a problem in the future, we can refactor at that point.